x-pack/plugins/data_enhanced/server/search/session/utils.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { createHash } from 'crypto';
+
+/**
+ * Generate the hash for this request so that, in the future, this hash can be used to look up
+ * existing search IDs for this request. Ignores the `preference` parameter since it generally won't
+ * match from one request to another identical request.
+ */
+export function createRequestHash(keys: Record<any, any>) {
+  const { preference, ...params } = keys;
+  return createHash(`sha256`).update(JSON.stringify(params)).digest('hex');
```

Comment:
`JSON.stringify` does not guarantee stable output. It'd be a good idea to use [`json-stable-stringify` instead](https://github.com/elastic/kibana/blob/5f781dcfecc9bcd96d9fa5ecfe5cd3e1cdedf094/x-pack%2Fplugins%2Flicensing%2Fserver%2Fplugin.ts#L11), so that you can get a deterministic hash for these requests.